src/runtime/omp/omp_queue.cpp

```diff
@@ -210,23 +232,20 @@ launch_kernel_from_so(omp_sscp_executable_object::omp_sscp_kernel *kernel,
   {
     // get page aligned local memory from heap
     static thread_local std::vector<char> local_memory;
-
-    // compiler/libkernel builtins assume that local mem is aligned to at least
-    // 512 byte boundaries
-    const auto local_mem_alignment = std::max(std::size_t{512}, get_page_size());
-    local_memory.resize(shared_memory + local_mem_alignment);
-    auto aligned_local_memory = reinterpret_cast<void *>(
-        next_multiple_of(reinterpret_cast<std::uint64_t>(local_memory.data()),
-                         local_mem_alignment));
-
+    static thread_local std::vector<char> internal_local_memory;
+    auto aligned_local_memory =
+        resize_and_strongly_align(local_memory, shared_memory);
+    auto aligned_internal_local_memory = resize_and_strongly_align(
+        internal_local_memory, local_size.size() * sizeof(uint64_t));
```

Comment:
Just to confirm if the size is sufficient?
In OpenMP we had: 128*1024 bytes (IIRC, this was due to `vec<double,16>`). We won't need this here?

Reply:
Yes, this will be fine, at least for now. SSCP group algorithms operate component-wise, so we will never need more than 64-bit per element (well, unless `int128_t` becomes a proper thing in SYCL). The motivation here is that we also need to support `marray` which has arbitrary number of components - so it makes sense to just handle scalar types component-wise.